ouroboros-consensus-diffusion/src/ouroboros-consensus-diffusion/Ouroboros/Consensus/NodeKernel.hs

```diff
@@ -426,208 +432,228 @@ forkBlockForging ::
 forkBlockForging IS{..} blockForging =
     forkLinkedWatcher registry threadLabel
     $ knownSlotWatcher btime
-    $ withEarlyExit_ . go
+    $ \currentSlot -> withRegistry (\rr -> withEarlyExit_ $ go rr currentSlot)
   where
     threadLabel :: String
     threadLabel =
         "NodeKernel.blockForging." <> Text.unpack (forgeLabel blockForging)
 
-    go :: SlotNo -> WithEarlyExit m ()
-    go currentSlot = do
-        trace $ TraceStartLeadershipCheck currentSlot
-
-        -- Figure out which block to connect to
-        --
-        -- Normally this will be the current block at the tip, but it may
-        -- be the /previous/ block, if there were multiple slot leaders
-        BlockContext{bcBlockNo, bcPrevPoint} <- do
-          eBlkCtx <- lift $ atomically $
-            mkCurrentBlockContext currentSlot
+    go :: ResourceRegistry m -> SlotNo -> WithEarlyExit m ()
```

Comment:
Please remove the change to the indentation level, so that this diff is much easier to review.